infra/blueprint-test/modules/tft/terraform.go

```diff
@@ -0,0 +1,248 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package tft provides a set of helpers to test Terraform modules/blueprints.
+package tft
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	gotest "testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/bpt"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/discovery"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/utils"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/mitchellh/go-testing-interface"
+	"github.com/stretchr/testify/assert"
+)
+
+// TFBlueprintTest implements bpt.Blueprint and stores information associated with a Terraform blueprint test.
+type TFBlueprintTest struct {
+	name        string                 // descriptive name for the test
+	fixtureName string                 // fixture name which will be used to compute TFDir
+	tfDir       string                 // directory containing Terraform configs
+	envVars     map[string]string      // variables to pass to Terraform as environment variables
+	setupPath   string                 // optional directory containing applied TF configs to import outputs as variables for the test
+	vars        map[string]interface{} // variables to pass to Terraform as flags
+	logger      *logger.Logger         // custom logger
+	t           testing.TB             // TestingT or TestingB
+}
+
+type option func(*TFBlueprintTest)
+
+func WithName(name string) option {
+	return func(f *TFBlueprintTest) {
+		f.name = name
+	}
+}
+
+func WithFixtureName(fixtureName string) option {
+	return func(f *TFBlueprintTest) {
+		f.fixtureName = fixtureName
+	}
+}
+
+func WithTFDir(tfDir string) option {
+	return func(f *TFBlueprintTest) {
+		f.tfDir = tfDir
+	}
+}
+
+func WithEnvVars(envVars map[string]string) option {
+	return func(f *TFBlueprintTest) {
+		f.envVars = envVars
+	}
+}
+
+func WithSetupPath(setupPath string) option {
+	return func(f *TFBlueprintTest) {
+		f.setupPath = setupPath
+	}
+}
+
+func WithVars(vars map[string]interface{}) option {
+	return func(f *TFBlueprintTest) {
+		f.vars = vars
+	}
+}
+
+func WithLogger(logger *logger.Logger) option {
+	return func(f *TFBlueprintTest) {
+		f.logger = logger
+	}
+}
+
+// Init sets defaults, validates and returns a TFBlueprintTest.
+func Init(t testing.TB, opts ...option) *TFBlueprintTest {
+	tft := &TFBlueprintTest{
+		name: fmt.Sprintf("%s TF Blueprint", t.Name()),
+		t:    t,
+	}
+	// apply options
+	for _, opt := range opts {
+		opt(tft)
+	}
+	// if no custom logger, set default based on test verbosity
+	if tft.logger == nil {
+		tft.logger = utils.GetLoggerFromT()
+	}
+	// one of fixture name or tfDir should be provided
```

Comment:
```suggestion
	// only one of fixture name or tfDir should be provided
```

Reply:
refactored this so we no longer have a case of both fixtureName and tfDir